src/main.test.js

```diff
@@ -23,40 +23,98 @@ function getNodeVersion() {
 }
 
 describe('Environmental Check', () => {
-  try {
-    childProcess.execSync('systeminfo');
+  let OS;
+  let nodeVersion;
+  let allActiveProgrammes;
+  let allProgrammes;
+  let listOfExtensions;
 
-    const allTasks = childProcess.execSync('tasklist').toString();
+  beforeAll(() => {
+    nodeVersion = getNodeVersion();
 
-    test('You should have Node.js of 14 version', () => {
-      const version = getNodeVersion();
+    listOfExtensions = childProcess.execSync(
+      'code --list-extensions'
+    ).toString();
 
-      expect(
-        (
-          version >= minVersionOfNode
-          && version < maxVersionOfNode
-        )
+    try {
+      childProcess.execSync('systeminfo');
+      OS = 'Windows';
+      allActiveProgrammes = childProcess.execSync('tasklist').toString();
+    } catch (error) {
+      try {
+        childProcess.execSync('lsb_release -a');
+        allProgrammes = childProcess.execSync('dpkg -l').toString();
+        OS = 'Linux';
+      } catch (e) {
+        OS = 'MacOS';
+      }
+    }
+  });
+
+  test('You should have Node.js of 14 version', () => {
+    expect(
+      (
+        nodeVersion >= minVersionOfNode
+          && nodeVersion < maxVersionOfNode
       )
-        .toBeTruthy();
-    });
+    )
+      .toBeTruthy();
+  });
 
-    test('You should have Git of 2.31.1 version or newer', () => {
-      const version = childProcess.execSync('git --version').toString();
+  test('You should have Git of 2.31.1 version or newer', () => {
+    const version = childProcess.execSync('git --version').toString();
 
+    if (OS === 'Windows') {
       expect((version.replace(/[^0-9]/g, '') >= minVersionOfGitOnWindows))
         .toBeTruthy();
-    });
-
-    test('You should have Bash Shell', () => {
-      const bashPath = childProcess.execSync('which bash').toString();
+    }
 
-      expect(!!bashPath)
+    if (OS !== 'Windows') {
+      expect((version.replace(/[^0-9]/g, '') >= minVersionOfGitOnMacAndLinux))
```

Comment:
duplicated
version.replace(/[^0-9]/g, '')
also you may replace 2 if statements with ternary `const minVersion = cond ? a : b`